pkg/serializer/func_queue.go

```diff
@@ -3,100 +3,50 @@
 
 package serializer
 
-import (
-	"context"
-	"fmt"
-)
-
-var (
-	// NoRetry always returns false independently of the number of retries.
-	NoRetry = func(int) bool { return false }
-)
-
-// WaitFunc will be invoked each time a queued function has returned an error.
-// nRetries will be set to the number of consecutive execution failures that
-// have occurred so far. The WaitFunc must return true if execution must be
-// retried or false if the function must be returned from the queue.
-type WaitFunc func(nRetries int) bool
-
-type queuedFunction struct {
-	f        func() error
-	waitFunc WaitFunc
-}
-
 type FunctionQueue struct {
-	queue  chan queuedFunction
+	queue  chan func() error
 	stopCh chan struct{}
+	err    error
 }
 
 // NewFunctionQueue returns a FunctionQueue that will be used to execute
 // functions in the same order they are enqueued.
-func NewFunctionQueue(queueSize uint) *FunctionQueue {
+func NewFunctionQueue() *FunctionQueue {
```

Comment:
nit: The comment here implies that multiple functions will be executed ("same order they are enqueued") but at this point only one function gets executed.

Arguably it's not a `FunctionQueue` any more since it just runs the first function that's sent to it.

These are cosmetic changes though, so feel free to either fix this up in an extra patch here or follow up separately.

Reply:
This `FunctionQueue` will be removed in subsequent work (see `replace with sync.Once` comment). Will scope this out in a few GH issues, as there's a lot to cover.

Marking as ready to merge.